src/mrack/outputs/ansible_inventory.py

```diff
@@ -200,12 +200,18 @@ def create_ansible_host(self, name):
 
         return host_info
 
+    def get_inventory_layout(self):
```

Comment:
I do not see much point in adding a custom inventory layout, we can add host groups using "groups:" already.

Reply:
Inventory layout purpose is not adding groups but rather grouping of groups, creating a hierarchy. This is useful in general preparation playbooks where plays can be limited to e.g. linux or windows. Or some other plays can be limited to very special groups.

Reply:
If we change the sssd-ci-containers playbooks, then it is not that important to stick to particular layout. However, being able to override the default values and clear them is still important so things don't get broken by changing the value in provisioning-config.

Reply:
I am afraid that we will have a huge definition of layout copypasted in metadata making it long and messy.
Maybe adding an option to have the layout in external file would be better.
The config then can have just one line with path referring to the layout/template.

Reply:
Not against, but it is not that far from what can be done now: kinda in similar spirit (overriding some file on defined location). You can now add a step (playbook, command or something else) in metadata after the `playbook: init/testrunner.yaml`. It can load the provisioning config, modify the layout and save the file.  Yes, it is slightly more complex than just replacing a file, but actually not that much as you still need some step to do it.

Nowdays, I'm more thinking about generic way how to extend + override provisioning config - some kind of clever merge. There might be generic stuff, useful for multiple jobs defined e.g. on upstream, downstream and project specific level.

Reply:
There is https://github.com/tanbro/pyyaml-include that gives you include for free.

Reply:
BTW: I got bitten in my ass by the magic done with the inventory twice now so I vote for extracting the layout from provisioning config to standalone file that can be included.
1. I used group ipa in my inventory and have machines marked a ipaserver and ipaclient (to select flavour) added to it unexpectedly.
2. A new group client was added resulting in name collision with a machine named client.

We could have layouts for different projects pre-defined in different files and have mrack-up with extra_vars to select from them or add a custom one.

Reply:
I think the way to go would be to allow override everything from provisioning-config in metadata through the config section (both merge and replace mechanisms should imho be supported) as we talked about in the office. Include makes sense to me and can be easily implemented by pyyaml-include. But I don't see any work related to this PR so I'm marking this as resolved. Please, reopen it if you think otherwise.